platform/mbed_rtc_time.cpp

```diff
@@ -94,24 +112,45 @@ time_t time(time_t *timer)
     if (_rtc_read != NULL) {
         t = _rtc_read();
     }
+    
+    tv->tv_sec = t;  
+        
+    _mutex->unlock();
 
-    if (timer != NULL) {
-        *timer = t;
+    // timezone struct is deprecated
+    if (tz) {
+        MBED_WARNING(MBED_MAKE_ERROR(MBED_MODULE_PLATFORM,  MBED_ERROR_CODE_INVALID_ARGUMENT), 
```

Comment:
Seems like a waste of code space. We've never implemented this function at all, so have no legacy behaviour ourselves, and POSIX says 'tz' is unspecified, so we can just ignore it. Returning "1" is inappropriate.